builtin/logical/database/backend.go

```diff
@@ -159,6 +160,8 @@ func (b *databaseBackend) Role(ctx context.Context, s logical.Storage, roleName
 		result.Statements = stmts
 	}
 
+	result.Statements.Revocation = strutil.RemoveEmpty(result.Statements.Revocation)
```

Comment:
You might want to move this to below the compatibility helper call below

Reply:
I specifically didn't as I figured the compatibility helper will then operate on valid data. Do you think that's the wrong approach?

Reply:
No, i think it's fine here. I just looked and the compatability helper will not set an empty string into the array.